_posts/2020-06-26-Making-IrishGen-SPARQL-Part-I.md

```diff
@@ -0,0 +1,549 @@
+The result is an order list going from most to least which shows each
+individual and their `:numChild` which is more than 1000 results.
+Thankfully, this is a very fast query at 68 milliseconds.
+Benchmarking software is a exhaustingly large subject so there is a
+very high variance in what the user may experience on their own
+computers due to _inter alia_ CPU L1 and L2 cache strategies, RAM
+clock rate, disk speed, query optimiser, and even more complicated
+events outside the control of the Triplestore.
+
+As the result is over 1000, we will only concern ourselves with the
+top five. We can do that progrmmatically by using the `limit` keyword.
+
+```sparql
+prefix irishRel: <http://example.com/earlyIrishRelationship.ttl#> 
+
+select ?x (max(?numChild) as ?maxNumChild) 
+from <tag:stardog:api:context:all>
+where {
+    ?x irishRel:numChild ?numChild
+} 
+group by ?x
+order by desc (?maxNumChild)
+limit 5
+```
+
+
+| ?x                                                                                                      | ?maxNumChild |
+| http://example.com/Laud_Misc_610/CGH/do_minigud_senchais_fer_muman.trig#Óengus                          | 48           |
+| http://example.com/Rawl_B502/de_genelogia_síl_ébir_16.trig#Óengus-e74701aa                              | 48           |
+| http://example.com/Rawl_B502/de_peritia_&_de_genealogis_dál_niad_cuirp_incipit.trig#CathairMár          | 33           |
+| http://example.com/Rawl_B502/de_peritia_&_de_genealogis_dál_niad_cuirp_incipit.trig#CathairMár-82bed530 | 30           |
+| http://example.com/LL/de_genelach_dáil_nia_corbb.trig#CathairMár                                        | 30           |
+
+The results of this raise an interesting point.  The reader will
+notice that Cathair Már appears three times in the results.  This
+means that there are _three_ different declared number of children for
+Cathair Már and they are not counted the same as `max` should return
+only the largest for a single individual.  To investigate this is
+outside the scope of this post but a few hypothesises suggest
+themselves.  First, that Cathair Már has differing numbers and the
+genealogists were themselves confused while taking from their own
+sources or constructing genealogies with competing interests.  Second,
+these are all the same person but they have not been linked together
+by `owl:sameAs` by the curators which would then match what is
+expected.  Third, there is a discrepancy between LL and Rawl B502 that
+is now apparent by looking at this query.  Whatever the actual
+underlying cause, the query demonstrates something that would need to
+be done painstakingly by hand with an even larger margin for error
+than using a Triplestore to do the calculation for the user.  Large
+aggregate questions about medieval Ireland can now be asked and
+answered with a greater degree of confidence.
+
+There is, of course, another method of counting by taking advantage of
```

Comment:
You need to make it much clearer that you are counting the number of children actually ascribed to people in the genealogies now, not merely returning the figures given explicitly in the genealogies.